pkg/controller/node/nodecontroller.go

```diff
@@ -232,7 +242,7 @@ func (nc *NodeController) monitorNodeStatus() error {
 		for node := range deleted {
 			glog.V(1).Infof("NodeController observed a Node deletion: %v", node)
 			nc.recordNodeEvent(node, "RemovingNode", fmt.Sprintf("Removing Node %v from NodeController", node))
-			nc.deleteNode(node)
+			nc.podEvictor.Add(node)
```

Comment:
(Can't comment on unchanged lines): When adding a new node you probably want to remove it from the evictor(s).